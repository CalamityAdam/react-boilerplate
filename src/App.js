import React, { useEffect, useState } from 'react';
import Col from 'react-bootstrap/Col';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import { Navigation } from './components/Navigation';
import { PokemonCard } from './components/PokemonCard';

function App() {
  const [pokemon, setPokemon] = useState([]);

  useEffect(() => {
    fetch('https://pokeapi.co/api/v2/pokemon/?limit=150')
      .then((res) => res.json())
      .then((data) => {
        setPokemon(data.results);
      })
      .catch((error) => {
        console.error(error);
      });
  }, []);

  return (
    <>
      <Navigation />

      <Container>
        <Row className='g-4'>
          {pokemon.map((poke) => (
            <Col key={poke.name}>
              <PokemonCard url={poke.url} name={poke.name} />
            </Col>
          ))}
        </Row>
      </Container>
    </>
  );
}

export { App };
