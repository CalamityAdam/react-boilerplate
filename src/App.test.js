/** @jest-environment jsdom */
import React from 'react';
import { act, render, screen } from '@testing-library/react';
import { App } from './App';
const initialFetch = window.fetch;

describe('App', () => {
  beforeEach(() => {
    window.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () =>
          Promise.resolve({
            results: [],
          }),
      })
    );
  });
  afterEach(() => {
    window.fetch = initialFetch;
  });

  it('Should render', async () => {
    await act(async () => {
      render(<App />);
    });

    const appContainer = screen.getByTestId('app');

    expect(appContainer).toBeInTheDocument();
  });
});
