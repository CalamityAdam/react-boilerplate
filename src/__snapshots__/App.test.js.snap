// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render 1`] = `
<body>
  <div>
    <nav
      class="navbar navbar-expand navbar-dark bg-dark sticky-top"
    >
      <div
        class="container"
      >
        <span
          class="navbar-brand"
        >
          Pokeverse
        </span>
        <div
          class="me-auto navbar-nav"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="/"
            href="/"
          >
            All Pokemon
          </a>
        </div>
      </div>
    </nav>
    <div
      class="container"
    >
      <div
        class="g-4 row"
      />
    </div>
  </div>
</body>
`;
